### PR TITLE
Reduce ATELIER runtime env coupling across subprocesses

### LINKED ISSUE #435: Reduce ATELIER_* runtime env coupling across subprocesses
Audit ATELIER_* environment variables injected into planner/worker subprocesses and replace project-sensitive runtime routing with explicit arguments/config-derived context. Preserve only session-local identity/context needed for active hook flows, and remove ambient env dependencies that can leak across projects.

Changeset strategy:
1) Inventory + classify vars: identity/context vs project-routing vs user defaults.;2) Remove/replace project-routing vars first (following at-zhz2 isolation fix).;3) Add launch-time env sanitizer + validation; convert remaining consumers to explicit params where practical.;4) Roll out with deprecation notices and tests.

Acceptance criteria:
Document every ATELIER_* variable currently set for subprocess execution, with owner and consumer paths.;Project-sensitive routing (repo/beads/provider selection) no longer depends on inherited ambient env; it is passed explicitly from resolved project context.;Hook/session identity vars are either removed or constrained to process-local launch environments with validation to prevent cross-project bleed.;Migration includes compatibility window, explicit deprecation warnings, and regression tests for multi-project concurrent use.

Local bead: at-pwlm

### PR DESCRIPTION
**Pull request by shaug**
# Summary

- Reduce ambient `ATELIER_*` runtime coupling for planner/worker/editor subprocesses by sanitizing inherited routing keys and rebuilding launch context explicitly.
- Preserve user-default runtime controls needed for diagnostics while dropping project-routing keys that can bleed across projects.

# Changes

- Added `src/atelier/runtime_env.py` and integrated it into subprocess env builders so inherited runtime-routing vars are dropped before launch.
- Added compatibility warnings for dropped ambient routing keys, including the documented removal date (`2026-07-01`).
- Added explicit runtime repo hint resolution and migrated affected flows away from ambient `ATELIER_PROJECT` routing assumptions.
- Added `docs/runtime-env-subprocess-inventory.md` with owner/consumer inventory for runtime `ATELIER_*` subprocess vars.
- Preserved `ATELIER_WORK_AGENT_TRACE` in `USER_DEFAULT_ENV_KEYS` so worker trace diagnostics still propagate through sanitized subprocess environments.
- Added regression coverage for env sanitization edge cases, including explicit empty env maps and trace-key propagation.

# Testing

- `just format`
- `just lint`
- `just test`

# Tickets

- Fixes #435

# Risks / Rollout

- Main behavior change is env sanitization: workflows relying on inherited runtime-routing keys now receive warnings and should migrate to launch-time project context.
- Compatibility fallback remains time-bounded with explicit deprecation messaging.

### CODE REVIEW

File: src/atelier/workspace.py [hunk 1]
```diff
@@ -15,7 +16,7 @@ def workspace_environment(
     base_env: Mapping[str, str] | None = None,
 ) -> dict[str, str]:
     """Build environment variables for workspace-aware subprocesses."""
-    env = dict(base_env or os.environ)
+    env, _removed = sanitize_subprocess_environment(base_env=base_env or os.environ)
```
Comment (shaug) [inline src/atelier/workspace.py]: `workspace_environment` now re-sanitizes `base_env`, but `base_env` is already built by `agents.agent_environment(...)` and includes the injected identity keys. As written, this second sanitize pass will drop `ATELIER_AGENT_ID` unless it is explicitly preserved. That can break subprocesses/skills that key off `ATELIER_AGENT_ID` for routing/ownership. Can we preserve identity keys here (e.g. pass `preserve_keys=("ATELIER_AGENT_ID",)` or sanitize only inherited ambient env before injection)?

File: src/atelier/runtime_env.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+_WARNING_SAMPLE_LIMIT = 6
+
+USER_DEFAULT_ENV_KEYS: frozenset[str] = frozenset(
+    {
+        "ATELIER_MODE",
+        "ATELIER_RUN_MODE",
+        "ATELIER_WATCH_INTERVAL",
+        "ATELIER_WORK_YES",
+        "ATELIER_PLAN_TRACE",
+        "ATELIER_WORK_TRACE",
```
[the remark this answers is not in the document]
Comment (shaug) [reply src/atelier/runtime_env.py]: Implemented in this update: `ATELIER_WORK_AGENT_TRACE` is now preserved in `USER_DEFAULT_ENV_KEYS`, and `tests/atelier/test_runtime_env.py` now asserts that `ATELIER_WORK_AGENT_TRACE=1` survives sanitization while runtime-routing keys are still removed.
